src/actix/api_key.rs

```diff
@@ -3,20 +3,36 @@ use std::future::{ready, Ready};
 use actix_web::body::{BoxBody, EitherBody};
 use actix_web::dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform};
 use actix_web::http::header::Header;
+use actix_web::http::Method;
 use actix_web::{Error, HttpResponse};
 use actix_web_httpauth::headers::authorization::{Authorization, Bearer};
-use constant_time_eq::constant_time_eq;
 use futures_util::future::LocalBoxFuture;
 
+use crate::common::auth::AuthKeys;
+
+const READ_ONLY_POST_PATTERNS: [&str; 11] = [
```

Comment:
would need to find a to somehow force this list updated, otherwise we will, for sure, forget to update it next time we introduce another API

Reply:
Do you think enum it'd work?

Reply:
I am not sure I understand how enum could help in this case

Reply:
Not sure if it's a good idea; I'm just throwing out brainstorming results to see our possibilities here.

Every solution I can think of requires us to set some parameters or new things manually.

The best option I can think about would be to add in the PR template to confirm that if new APIs were added, we should add the pattern here (as well on RPC_PATHS, if required)

Reply:
One option that I thought about would be to make the APIs return their patterns, but this is sub-optimal since it'd increase complexity